app/presenters/describe_repair.rb

```diff
@@ -0,0 +1,35 @@
+class DescribeRepair
+  attr_reader :form
+
+  def initialize(form_params: {}, details:, answers:)
+    @form_params = form_params
+    @answers = answers
+    @details = details
+  end
+
+  def partial
+    if diagnosed?
+      'anything_else'
+    else
+      @details || 'describe_problem'
+    end
+  end
+
+  def form
+    @form ||= build_form(@form_params)
+  end
+
+  private
+
+  def diagnosed?
+    !RepairParams.new(@answers).sor_code.nil?
```

Comment:
How about this instead:

`RepairParams.new(@answers).sor_code.present?`

Reply:
That is indeed clearer. I thought this wouldn't be available in this class, but I guess ActiveModel includes it. 

Since we're asking this question of RepairParams in two places, I've created a `diagnosed?` method: that also makes us more flexible if we wanted to redefine what it means to be 'diagnosed` in future.